app/test_main.py

```diff
@@ -1 +1,39 @@
-# write your code here
+from pytest import MonkeyPatch, mark
+
+
+from app.main import can_access_google_page
+
+GOOGLE_URL = "Some google url"
+NOT_GOOGLE_URL = "Not google url"
+
+
+@mark.parametrize(
+    "url, url_validation,internet,result",
+    [
+        (GOOGLE_URL, False, True, "Not accessible"),
+        (GOOGLE_URL, True, True, "Accessible"),
+        (NOT_GOOGLE_URL, False, True, "Not accessible"),
+        (GOOGLE_URL, True, True, "Accessible"),
+        (NOT_GOOGLE_URL, False, False, "Not accessible"),
+        (GOOGLE_URL, True, False, "Not accessible"),
+    ],
```

Comment:
why do u need different "urls" for monkeypatch ?
it seems to me, u can make here only 4 tests for all cases